tensorflow/python/ops/cond_v2.py

```diff
@@ -136,11 +136,11 @@ def cond_v2(pred, true_fn, false_fn, name="cond"):
     # correct output structure
     tensors = tuple(array_ops.identity(t) for t in tensors)
 
-    result = tuple(tensors[:num_cond_outputs])
-    if len(result) == 1:
-      return result[0]
-    else:
-      return result
+    # Packing output tensors in the same nested structure as the true and false
+    # functions return
+    result = nest.pack_sequence_as(structure=true_graph.structured_outputs, 
+                                   flat_sequence=tensors[:num_cond_outputs])
```

Comment:
Didn't you have a check in _check_same_outputs that true_graph and false_graph have the same outputs?

Reply:
@skye My apologies there was another versioning error on my end. Should be fixed with the latest commit.